web/src/main/java/org/phoenixctms/ctsms/web/model/LazyDataModelBase.java

```diff
@@ -28,6 +28,22 @@ protected Long getPageId(IDVO idvo) {
 		return idvo.getId();
 	}
 
+	@Override
+	public void setRowIndex(int rowIndex){
+		/*
+		 * the following is a workaround to avoid div by zero in case of pageSize=0
+		 * (possible bug in pf)
+		 * this.rowIndex = rowIndex == -1 ? rowIndex : (rowIndex % pageSize);
+		 */
+		if (rowIndex == -1 || getPageSize() == 0) {
+			super.setRowIndex(-1);
+		}
+		else {
+			super.setRowIndex(rowIndex % getPageSize());
```

Comment:
ok legacy code was

```
    public void setRowIndex(int rowIndex) {
        this.rowIndex = rowIndex == -1 ? rowIndex : (rowIndex % pageSize);
    }
```

so it should be 


```
super.setRowIndex(rowIndex);
```

right now it only works because % is idempotent

Reply:
of course, that's right